buzsaki_lab_to_nwb/valero/stimulilaserinterface.py

```diff
@@ -0,0 +1,152 @@
+        if device_metadata["name"] not in nwbfile.devices:
+            neurolight_probe = nwbfile.create_device(**device_metadata)
+        else:
+            neurolight_probe = nwbfile.devices[device_metadata["name"]]
+        # Create the sites
+        site_description = f"Microled site in Neurolight probe. Microscopic LED 10 x 15 µm each, 3 per shank. Each μLED has an emission area of 150 μm2"
+        location = "dorsal right hippocampus (antero-posterior 2.0 mm, mediolateral 1.5 mm, dorsoventral 0.6 mm)"
+
+        micro_led_ids_to_site = dict()
+        micro_led_ids = np.unique(pulse_micro_led)
+
+        for id in micro_led_ids:
+            optogenetic_site = OptogeneticStimulusSite(
+                name=f"OptogeneticStimulusSite{id}",
+                device=neurolight_probe,
+                description=site_description,
+                excitation_lambda=460.0,  # nm
+                location=location,  # TODO find the mapping for precise location per site if possible
+            )
+            micro_led_ids_to_site[id] = optogenetic_site
+            nwbfile.add_ogen_site(optogenetic_site)
+
+        # Create the stimulus series
+        for id in micro_led_ids:
+            site_intervals = pulse_intervals[pulse_micro_led == id]
+            site_amplitudes = pulse_amplitude[pulse_micro_led == id]
+
+            # Assume from the trapezoidal profile that the decay time is the same as the rise time
+            pulse_start_time = site_intervals[:, 0]
+            amplitude_at_start = np.zeros_like(pulse_start_time)
+
+            raise_time = 0.001  # 1 ms
+            rise_to_max_time = pulse_start_time + raise_time
+            amplitude_at_max = site_amplitudes
+
+            pulse_stop_time = site_intervals[:, 1]
+            start_decaying_time = pulse_stop_time - raise_time
+            amplitude_start_decaying = site_amplitudes
+            ampltidue_stop_time = np.zeros_like(start_decaying_time)
+
+            timestamps = np.vstack((pulse_start_time, rise_to_max_time, start_decaying_time, pulse_stop_time))
+            data = np.vstack((amplitude_at_start, amplitude_at_max, amplitude_start_decaying, ampltidue_stop_time))
+
+            site_timestamps = timestamps.T.flatten()
+            site_data = data.T.flatten()
+
+            optogenetic_series_description = (
+                "μLEDs were controlled with current (2-4.5 μA generating 0.02-0.1μW of total light power;"
+                "ref (15)) provided by a 12-channel current generator (OSC1Lite, NeuroNex Michigan Hub)"
+                "driven by an Arduino, which delivered trapezoid (1ms rise time)"
+                "blue light (centered emission at 460 nm, emission surface area = 150 mm2) 20 ms pulses at"
+                "random sites with a randomly variable (40-60ms) offset"
+            )
+            optogenetic_site = micro_led_ids_to_site[id]
+            optogenetic_series = OptogeneticSeries(
+                name=f"Stimuli from microLED site {id}",
```

Comment:
Looks like this has the same problem with spaces

Reply:
Thanks, Cody, corrected it.